indexbuilder.go

```diff
@@ -559,3 +514,61 @@ func (b *IndexBuilder) branchMask(br string) uint64 {
 	}
 	return 0
 }
+
+type DocChecker struct {
```

Comment:
minor: now that we are revisiting this API, its kinda weird its an exported API in the zoekt package and is only ever used by the build package. Feels like it should be private struct inside of the build pkg.

Reply:
Update: This will require a large-ish refactor, so I won't tackle it in this PR.